### ISSUE TITLE
kamatera cloud provider unit tests are flaky

### ISSUE THREAD

**Issue by x13n** [kubernetes/autoscaler#5134]
**Which component are you using?**:

cluster-autoscaler

**What version of the component are you using?**:

HEAD commit from master

Component version:

**What k8s version are you using (`kubectl version`)?**:

n/a

**What environment is this in?**:

unit tests

**What did you expect to happen?**:

Tests should pass.

**What happened instead?**:

```
--- FAIL: TestNodeGroup_Others (0.01s)
    kamatera_node_group_test.go:285: 
        	Error Trace:	kamatera_node_group_test.go:285
        	Error:      	Not equal: 
        	            	expected: "node group ID: ng1 (min:1 max:7)\ninstance ID: 2773c4e7122d4bc58694ba4b275cfc0e state: Running powerOn: false\ninstance ID: 131a62b825a14f9c99335cd2d1e05ad5 state: Running powerOn: false\ninstance ID: bfccc7cc5ceb4011b32954c2d59186cb state: Running powerOn: false"
        	            	actual  : "node group ID: ng1 (min:1 max:7)\ninstance ID: 131a62b825a14f9c99335cd2d1e05ad5 state: Running powerOn: false\ninstance ID: bfccc7cc5ceb4011b32954c2d59186cb state: Running powerOn: false\ninstance ID: 2773c4e7122d4bc58694ba4b275cfc0e state: Running powerOn: false"
        	            	
        	            	Diff:
        	            	--- Expected
        	            	+++ Actual
        	            	@@ -1,4 +1,4 @@
        	            	 node group ID: ng1 (min:1 max:7)
        	            	-instance ID: 2773c4e7122d4bc58694ba4b275cfc0e state: Running powerOn: false
        	            	 instance ID: 131a62b825a14f9c99335cd2d1e05ad5 state: Running powerOn: false
        	            	 instance ID: bfccc7cc5ceb4011b32954c2d59186cb state: Running powerOn: false
        	            	+instance ID: 2773c4e7122d4bc58694ba4b275cfc0e state: Running powerOn: false
        	Test:       	TestNodeGroup_Others
FAIL
FAIL	k8s.io/autoscaler/cluster-autoscaler/cloudprovider/kamatera	8.148s
```

**How to reproduce it (as minimally and precisely as possible)**:

Run tests.

**Anything else we need to know?**:

The test relies on specific order of map entries, which is not deterministic.

/assign @OriHoch

**Comment by x13n** [comment]
Looks like I cannot assign the issue. However, @OriHoch - can you fix the tests?

**Comment by OriHoch** [comment]
thanks, fixed in #5137
